src/site/stages/build/drupal/metalsmith-drupal.js

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable no-param-reassign */
+
+const fs = require('fs-extra');
+const path = require('path');
+const chalk = require('chalk');
+
+const ENVIRONMENTS = require('../../../constants/environments');
+const getApiClient = require('./api');
+const GET_ALL_PAGES = require('./get-all-pages.graphql');
+
+const DRUPAL_CACHE_FILENAME = 'drupal.json';
+
+// If "--pull-drupal" is passed into the build args, then the build
+// should pull the latest Drupal data.
+const PULL_DRUPAL_BUILD_ARG = 'pull-drupal';
+
+const ENABLED_ENVIRONMENTS = new Set([
+  ENVIRONMENTS.LOCALHOST,
+  ENVIRONMENTS.VAGOVDEV,
+]);
+
+const DRUPAL_COLORIZED_OUTPUT = chalk.rgb(73, 167, 222);
+
+// eslint-disable-next-line no-console
+const log = message => console.log(DRUPAL_COLORIZED_OUTPUT(message));
+
+function pipeDrupalPagesIntoMetalsmith(contentData, files) {
+  const {
+    data: {
+      nodeQuery: { entities: pages },
+    },
+  } = contentData;
+
+  let drupalIndexPage =
+    '<h1>The following pages were provided by Drupal:</h1><ol>\n';
+
+  for (const page of pages) {
+    const {
+      entityUrl: { path: drupalPagePath },
+    } = page;
+
+    const jsonPath = `drupal${drupalPagePath}.json`;
+
+    drupalIndexPage += `<li><a href="/${jsonPath}">${drupalPagePath}</a></li>`;
+
+    files[jsonPath] = {
+      ...page,
+      contents: Buffer.from(JSON.stringify(page, null, 4)),
+    };
+  }
+
+  drupalIndexPage += '</ol>';
+
+  files['drupal/index.html'] = {
+    contents: Buffer.from(drupalIndexPage),
+  };
+
+  log('Drupal index page written to /drupal.');
+}
+
+async function loadDrupal(buildOptions) {
+  const drupalCache = path.join(
+    buildOptions.cacheDirectory,
+    DRUPAL_CACHE_FILENAME,
+  );
+  const isDrupalAvailableInCache = fs.existsSync(drupalCache);
+
+  let shouldPullDrupal = buildOptions[PULL_DRUPAL_BUILD_ARG];
+  let drupalPages = null;
+
+  if (!isDrupalAvailableInCache) {
+    log('Drupal content unavailable in cache');
+    shouldPullDrupal = true;
+  }
+
+  if (shouldPullDrupal) {
+    log('Attempting to load Drupal content from API...');
+
+    const contentApi = getApiClient(buildOptions);
+
+    drupalPages = await contentApi.query({ query: GET_ALL_PAGES });
+
+    if (buildOptions.buildtype === ENVIRONMENTS.LOCALHOST) {
+      const serialized = Buffer.from(JSON.stringify(drupalPages, null, 2));
+      fs.ensureDirSync(buildOptions.cacheDirectory);
+      fs.writeFileSync(drupalCache, serialized);
+    }
+  } else {
+    log('Attempting to load Drupal content from cache...');
+    log(`To pull latest, run with "--${PULL_DRUPAL_BUILD_ARG}" flag.`);
+
+    // eslint-disable-next-line import/no-dynamic-require
+    drupalPages = require(drupalCache);
+  }
+
+  log('Drupal successfully loaded!');
+  return drupalPages;
+}
```

Comment:
I did go ahead and add the cache, because we want to see this built to Dev and if we were relying on a build arg, we would have had to add that flag into the `vagovdev` build command, or do something in `options.js` to assign that flag based on buildtype. The cache alongside the environment feature flag is a better longterm solution. 

This works by -

1. The first time running, it will hit the Drupal API. On localhost, the response will be cached into `vets-website/.cache/drupal.json`
2. On subsequent builds, it'll read from that cache file unless the flag `--pull-drupal` is passed, in which case it'll do Step 1 again.

There isn't any timestamping at this time, but that'll be a good improvement.

## Running without cache or with --pull-drupal flag 
`npm run watch -- --pull-drupal`
![image](https://user-images.githubusercontent.com/1915775/50863952-7c0bf700-136e-11e9-8f0d-49aab6f8a124.png)


## Running with .cache/drupal.json
`npm run watch`
![image](https://user-images.githubusercontent.com/1915775/50863917-5bdc3800-136e-11e9-82a9-cbaf5f7dd0b3.png)

Is this output too much?

Reply:
I think that's ok. We need to make sure we stay on top of any issues that come up with hitting the Drupal api by local devs, but that shouldn't be too bad.